### PR TITLE
Initial work for deployment source

### PR DESCRIPTION
**Pull request by hpsniper**
Supply build and repository details as properties to mabl deployments. Here is an example output using subversion as the repository:

```
{
  "properties": {
    "build_plan_result_url": {
      "stringValue": "http://127.0.0.1:9090/job/Subversion/5/display/redirect"
    },
    "deployment_origin": {
      "stringValue": "mabl-jenkins-plugin/0.0.10-SNAPSHOT"
    },
    "repository_name": {
      "stringValue": "int_test_svn"
    },
    "build_plan_id": {
      "stringValue": "Subversion"
    },
    "repository_url": {
      "stringValue": "https://svn.riouxsvn.com/int_test_svn"
    },
    "repository_revision_number": {
      "stringValue": "6"
    },
    "build_plan_number": {
      "stringValue": "5"
    },
    "build_plan_name": {
      "stringValue": "Subversion"
    }
  }
}
```

### CODE REVIEW

File: src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java [hunk 1]
```diff
@@ -134,6 +136,21 @@ private FilePath getOutputFileLocation(AbstractBuild<?, ?> build) {
         return fp;
     }
 
+    private EnvVars getEnvironmentVars(AbstractBuild<?, ?> build, BuildListener listener) {
+        final PrintStream outputStream = listener.getLogger();
+        EnvVars environmentVars = new EnvVars();
+        try {
+            environmentVars = build.getEnvironment(listener);
+        } catch (IOException e) {
+            outputStream.println(e.getMessage());
```
Comment (twistedpair) [inline src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java]: We could add some logging detail like `Error gathering environmental variables`.

File: src/main/java/com/mabl/integration/jenkins/Converter.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        for(String property : possibleProperties) {
+            if(vars.containsKey(property)) {
+                return vars.get(property);
+            }
+        }
+
+        return null;
+    }
+
+    private static String getRepositoryName(String repoUrl) {
```
Comment (twistedpair) [inline src/main/java/com/mabl/integration/jenkins/Converter.java]: Any tests around these? I imagine odd edge cases with possible unusual repo names.
